File: src/com/Torvald/Terrarum/UserInterface/UICanvas.java
package com.Torvald.Terrarum.UserInterface;

import org.newdawn.slick.GameContainer;
import org.newdawn.slick.Graphics;
import org.newdawn.slick.Input;

import java.util.LinkedList;

/**
 * Created by minjaesong on 15-12-31. <br>
 */
public interface UICanvas {

    int getWidth();

    int getHeight();

    void update(GameContainer gc, int delta_t);

    void render(GameContainer gc, Graphics g);

    void processInput(Input input);

}

File: src/com/Torvald/Terrarum/UserInterface/UICanvas.kt
package com.Torvald.Terrarum.UserInterface

import org.newdawn.slick.GameContainer
import org.newdawn.slick.Graphics
import org.newdawn.slick.Input

/**
 * Created by minjaesong on 16-03-14.
 */
interface UICanvas {

    var width: Int?
    var height: Int?

    fun update(gc: GameContainer, delta_t: Int)

    fun render(gc: GameContainer, g: Graphics)

    fun processInput(input: Input)

}